### PR TITLE
MIP-9

### PR DESCRIPTION
**Pull request by n00m4d**
# What
This PR adds canopy size check to CreateTree instruction as described in [MIP-9](https://github.com/metaplex-foundation/mip/blob/main/mip-9.md)

### CODE REVIEW

File: programs/bubblegum/program/src/error.rs [hunk 1]
```diff
@@ -86,6 +86,8 @@ pub enum BubblegumError {
     CreatorDidNotUnverify,
     #[msg("Only NonFungible standard is supported")]
     InvalidTokenStandard,
+    #[msg("Canopy size should be set bigger for this tree")]
+    CanopySizeMismatch,
```
Comment (danenbm) [inline programs/bubblegum/program/src/error.rs:90]: Maybe we could call this `InvalidCanopySize` as its not really mismatched, its more just out of spec.  What do you think?

File: programs/bubblegum/program/src/processor/create_tree.rs [hunk 2]
```diff
@@ -54,3 +70,40 @@ pub(crate) fn create_tree(
...
+
+    let merkle_tree_size = merkle_tree_get_size(&header)?;
+
+    let (_tree_bytes, canopy_bytes) = rest.split_at(merkle_tree_size);
+
+    let required_canopy = if max_depth > MAX_ACC_PROOFS_SIZE {
+        max_depth - MAX_ACC_PROOFS_SIZE
+    } else {
+        0
+    };
```
Comment (danenbm) [inline programs/bubblegum/program/src/processor/create_tree.rs:99]: Note: This is fine and its readable but wanted to let you know you could use `saturating_sub` to make this a single line.

File: programs/bubblegum/program/tests/utils/context.rs [hunk 3]
```diff
@@ -178,12 +182,34 @@ impl BubblegumTestContext {
...
+        let payer = self.payer();
+        let mut tree = Tree::<MAX_DEPTH, MAX_BUFFER_SIZE>::with_creator_and_canopy(
+            &payer,
+            Some(canopy),
+            self.client(),
+        );
+        tree.alloc(&payer).await?;
+        if is_public {
+            tree.create_public(&payer).await?;
+        }
```
Comment (danenbm) [inline programs/bubblegum/program/tests/utils/context.rs:209]: Do you need `else { tree.create(&payer).await?; ]` here?

**Review by danenbm — CHANGES_REQUESTED** [review]
Thank you for doing this so quickly!  Changes look good, but left a couple comments.  Also we will need to regenerate the clients, both the Umi js client and the js-solita one.

File: programs/bubblegum/program/tests/utils/context.rs (hunk 3 above)
Comment (n00m4d) [reply programs/bubblegum/program/tests/utils/context.rs]: yep, missed that.
thanks!

**Review by danenbm — APPROVED** [review]
LGTM!

**Comment by danenbm** [comment]
I fixed the CI issue by deleting the Cargo.lock file for the Rust client.  This is a known issue that when we do cargo release, for some reason it downgrades the borsh version, causing a mismatch between borsh and kaigan packages.

Also i regenerated the solita JS SDK in case anyone is still using it.  I was not able to get the Solita tests to run (they don't run in CI either).  They still use yarn.   I went into js-solita directory and did:
```
yarn install
yarn build
yarn test
```
but the tests hang, I think with the airdrop.

Unless you can figure out how to run the solita tests manually, I think since we are just adding the one error code to Solita SDK and the Solita SDK is in process of being deprecated, its probably fine to leave them.

**Pull request merged**